etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiOrderConverter.java

```diff
@@ -168,6 +167,26 @@ private MessageHeader createMessageHeader() {
         return messageHeader;
     }
 
+    @Override
+    public Order<?> addEtorProcessingTag(Order<?> message) {
+        var hapiOrder = (Order<Bundle>) message;
+        var messageBundle = hapiOrder.getUnderlyingOrder();
+
+        var messageHeaderOptional =
+                HapiHelper.resourcesInBundle(messageBundle, MessageHeader.class).findFirst();
+        if (messageHeaderOptional.isPresent()) {
+            var messageHeader = messageHeaderOptional.get();
+            var meta = messageHeader.hasMeta() ? messageHeader.getMeta() : new Meta();
```

Comment:
Could we add a Meta constructor that takes a Coding object so we don't have to call setMeta after?

Reply:
Good suggestion. The `Meta` class is from the fhir library. If we want to overload its constructor, we will have to submit the request in their [repo](https://github.com/hapifhir/hapi-fhir)